lib/remap-lang/src/expression/function.rs

```diff
@@ -23,12 +23,18 @@ pub enum Error {
 
     #[error(r#"error for argument "{0}""#)]
     Argument(String, #[source] expression::argument::Error),
+
+    #[error(r#"cannot mark infallible function as "abort on error", remove the "!" signature"#)]
```

Comment:
I might expand this error to actually show them something like:

```
cannot mark infallible function as "abort on error", remove the "!' signature like: `parse_json!` -> `parse_json`
```

Reply:
Good suggestion. I'll add this to the issue that's tracking improved error handling

Reply:
Tracking in https://github.com/timberio/vector/issues/4803.